partiql-ast/src/main/kotlin/org/partiql/ast/normalize/NormalizeSelect.kt

```diff
@@ -0,0 +1,252 @@
+                    val alias = it.asAlias ?: error("The AT alias should be present. This wasn't normalized.")
+                    buildSimpleStruct(it.expr, alias.symbol)
+                }
+                val byAliasItem = byAlias?.simple()?.let {
+                    val alias = it.asAlias ?: error("The AT alias should be present. This wasn't normalized.")
+                    buildSimpleStruct(it.expr, alias.symbol)
+                }
+                listOfNotNull(
+                    buildCaseWhenStruct(asAlias.star(i).expr, i),
+                    atAliasItem,
+                    byAliasItem
+                )
+            }
+            selectValue {
+                constructor = exprCall {
+                    function = identifierSymbol("TUPLEUNION", Identifier.CaseSensitivity.SENSITIVE)
+                    args.addAll(tupleUnionArgs)
+                }
+                setq = select.setq
+            }
+        }
+
+        private fun visitSelectProjectWithProjectAll(node: Select.Project): AstNode = ast {
+            val tupleUnionArgs = node.items.mapIndexed { index, item ->
+                when (item) {
+                    is Select.Project.Item.All -> buildCaseWhenStruct(item.expr, index)
+                    is Select.Project.Item.Expression -> buildSimpleStruct(
+                        item.expr,
+                        item.asAlias?.symbol
+                            ?: error("The alias should've been here. This AST is not normalized.")
+                    )
+                }
+            }
+            selectValue {
+                setq = node.setq
+                constructor = exprCall {
+                    function = identifierSymbol("TUPLEUNION", Identifier.CaseSensitivity.SENSITIVE)
+                    args.addAll(tupleUnionArgs)
+                }
+            }
+        }
+
+        @OptIn(PartiQLValueExperimental::class)
+        private fun visitSelectProjectWithoutProjectAll(node: Select.Project): AstNode = ast {
+            val structFields = node.items.map { item ->
+                val itemExpr = item as? Select.Project.Item.Expression ?: error("Expected the projection to be an expression.")
+                exprStructField(
+                    name = exprLit(stringValue(itemExpr.asAlias?.symbol!!)),
+                    value = item.expr
+                )
+            }
+            selectValue {
+                setq = node.setq
+                constructor = exprStruct {
+                    fields.addAll(structFields)
+                }
+            }
+        }
+
+        @OptIn(PartiQLValueExperimental::class)
+        private fun buildCaseWhenStruct(expr: Expr, index: Int): Expr.Case {
+            return ast {
+                exprCase {
+                    branches.add(
+                        exprCaseBranch(
+                            condition = exprIsType(expr, typeStruct()),
+                            expr = expr
+                        )
+                    )
```

Comment:
Sometimes the `+=` syntax on mutable lists can reduce nesting
```suggestion
                    branches += exprCaseBranch(
                        condition = exprIsType(expr, typeStruct()),
                        expr = expr
                    )
```